cmdtest.go

```diff
@@ -691,6 +691,25 @@ func fixedArgBuiltin(nargs int, f func([]string) ([]byte, error)) CommandFunc {
 	}
 }
 
+func rename(oldpath, newpath string) error {
+	// NOTE: This is used in place of os.Rename because it's
+	// able to rename files between different mount points.
+	src, err := os.Open(oldpath)
+	if err != nil {
+		return err
+	}
+	dst, err := os.Create(newpath)
+	if err != nil {
+		return err
+	}
+	defer dst.Close()
+	defer src.Close()
```

Comment:
Nit: this should be moved up before the `os.Create` so that src is closed if `os.Create` fails.